astropy/coordinates/sky_coordinate.py

```diff
@@ -1904,6 +1905,10 @@ def from_name(cls, name, frame='icrs', parse=False):
             in this way may differ from the database coordinates by a few
             deci-arcseconds, so only use this option if you do not need
             sub-arcsecond accuracy for coordinates.
+        cache : bool, optional
+            Determines whether to cache the results or not.
+        overwrite_cache : bool, optional
```

Comment:
I don't understand the need to have two different keywords to do this. Why not ask user to use `cache='update'` directly like in `get_icrs_coordinates` above?

Reply:
Sigh, fine. I just hate string arguments like this so much!!

Reply:
Maybe we can slowly move to `Enum` - https://docs.python.org/3/library/enum.html